packages/std/src/results/subcall.rs

```diff
@@ -62,9 +62,16 @@ pub struct SubcallResponse {
     pub data: Option<Binary>,
 }
 
-/// a full sdk event
+/// A full Cosmos SDK event as documented in
+/// https://docs.cosmos.network/v0.42/core/events.html.
+///
+/// This version uses string attributes (similar to
+/// https://github.com/cosmos/cosmos-sdk/blob/v0.42.5/proto/cosmos/base/abci/v1beta1/abci.proto#L56-L70),
+/// which then get magically converted to bytes for Tendermint somewhere between
+/// the Rust-Go interface, JSON deserialization and the `NewEvent` call in Cosmos SDK.
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
 pub struct Event {
+    /// The event type. This is renamed to "kind" because "type" is reserved in Rust. This sucks, we know.
```

Comment:
Did you know you can escape reserved keywords by adding the `r#` prefix? Should we change `kind` to `r#type`?

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=87c3f73722909d3be66065512fdc016d

Reply:
Came across it be never used it really. This would mean in all places there the property is used you need to add the prefix, also here:

```rust
println!("{}", foo.r#type);
```

This does not seem to be nicer than what we currently have.

Reply:
The Wasmer codebases uses "ty" a lot as an approviation of type. I don't think this is great either.

Reply:
I think I've seen `ty` used in some other codebase too. But fair enough.

Reply:
I'm definitly open to change this, but it is somhow hard. Ooptions on the table:
1. Use a different word, like "kind"
2. Use an appreviation, like "ty"
3. Add prefix or suffix: `_type`/`type_`
4. Use escaping syntax `r#type`

Looking at this, I have a slight tendency to `ty` or `type_` is this at least is very close in the mental model.

@ethanfrey thoughts on this?

Reply:
I prefer using serde for the renaming (over `r#type`, but that is an interesting point to know)
I don't like the underscores too much.

1 and 2 are both fine for me. Happy to change this to `ty`.

Reply:
https://github.com/CosmWasm/cosmwasm/issues/967